pygfx/renderers/wgpu/meshshader.py

```diff
@@ -210,8 +246,18 @@ def code_vertex(self):
 
             // Select what face we're at
             let index = i32(in.vertex_index);
-            let face_index = index / 3;
-            var sub_index = index % 3;
+            let face_index = index / {{indexer}};
+            let sub_index_ori = index % {{indexer}};
+            var sub_index = index % {{indexer}};
+            
+            // for quads assuming the vertices are oriented, the triangles are 0 1 2 and 0 2 3
+            if sub_index == 3 {
+                sub_index = 0;
+                } else if sub_index == 4 {
+                sub_index = 2;
+                } else if sub_index == 5 {
+                sub_index = 3;
+                }
```

Comment:
Its good practice to avoid if-statements in shaders. Maps are usually more performant. And in this case the code is actually simpler :)

Also, lets use templating to only use this code for quads.

```suggestion
            // for quads assuming the vertices are oriented, the triangles are 0 1 2 and 0 2 3
            $$ if indexer == 3
            var quad_map = array<f32,6>(0, 1, 2, 0, 2, 3);
            sub_index = quad_map[sub_index];
            $$ endif
```

Reply:
corrected this to 
```
   $$ if indexer == 6
            var quad_map = array<i32,6>(0, 1, 2, 0, 2, 3);
```